reflex/state.py

```diff
@@ -1795,8 +1845,13 @@ def dict(
             self.dirty_vars.update(self._always_dirty_computed_vars)
             self._mark_dirty()
 
+        def dictify(value: Any):
+            if dataclasses.is_dataclass(value) and not isinstance(value, type):
+                return dataclasses.asdict(value)
+            return value
+
         base_vars = {
-            prop_name: self.get_value(getattr(self, prop_name))
+            prop_name: dictify(self.get_value(getattr(self, prop_name)))
```

Comment:
This change caused issues for us, because `dataclasses.asdict` raises exceptions when used with sqlalchemy mapped dataclasses. Do we even need this?
If we need it, can we add `and not has_serializer(type(value))` to the condition in `dictify` to fix it?

cc @adhami3310 @masenf

Reply:
It should work without it as we have now logic covering dataclasses inside of serialize. I'll make a PR soon.

Reply:
#4141